sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -994,3 +995,121 @@ mod test_topology_manager_policy {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesClusterDNSIp represents the --cluster-dns settings for kubelet.
+///
+/// The default is determined via information from IMDS in AWS.
+/// This model allows the value to be either a list of IPs, or a single IP string
+/// for backwards compatibility.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum KubernetesClusterDNSIp {
+    Scalar(IpAddr),
+    Vector(Vec<IpAddr>),
+}
+
+impl KubernetesClusterDNSIp {
+    pub fn iter<'a>(&'a self) -> Box<dyn Iterator<Item = &'a IpAddr> + 'a> {
+        match self {
+            Self::Scalar(inner) => Box::new(std::iter::once(inner)),
+            Self::Vector(inner) => Box::new(inner.iter()),
+        }
+    }
+
+    pub fn into_iter(self) -> impl Iterator<Item = IpAddr> {
+        match self {
+            Self::Scalar(inner) => vec![inner],
+            Self::Vector(inner) => inner,
+        }
+        .into_iter()
+    }
+
+    pub fn first(&self) -> Option<&IpAddr> {
+        self.iter().next()
+    }
```

Comment:
Why add a special `first` method vs. making callers write `iter().next()` ? It seems like giving them the iterator and letting them use any of the trait methods is more powerful.